Monk Сlass

zumbak04 opened the pull request:
<!--
A basic changelog, goes to patch notes of the next version, so should be as short and informative as possible.
-->
## Changelog:
- Added the Monk class with 3 abilities:
  - Heal Wounds.
  - Wave Soothing Mist to heal depression and stress.
  - Heal Army combat event to heal your troops and increase their morale.

<!--
If you need to explain something to testers. Otherwise, delete this part.
-->
# How to test:
For now, everyone can be monks. Later I will limit this to the Chil religion group.

A few commands:
- `add_trait class_monk_10` to become a monk.
- `event WCCLS.10500` to fire the Heal Army combat event.
- `event WCCLS.5` to restore Mana.

Try to become a monk without using the command: both via the Intrigue decision and right-clicking other monks to become your mentor.

Try the abilities and tell your opinion.

arithon commented on localisation/00_zuwc_united_localisation.csv:
```diff
@@ -1442,7 +1442,7 @@ has_3_4_physical_class_trigger_tooltip;Physical class level is 3 or 4\n;Classe p
 has_7_8_physical_class_trigger_tooltip;Physical class level is 7 or 8\n;Classe physique de niveau 7 ou 8\n;Phyische Klassenstufe betr�gt 7 oder 8\n;;;;;;;;;;;x
 has_9_10_physical_class_trigger_tooltip;Physical class level is 9 or 10\n;Classe physique de niveau 9 ou 10\n;Phyische Klassenstufe betr�gt 9 oder 10\n;;;;;;;;;;;x
-EVTDESC_WCCLS_105;You fought in the heat of battle alongside your soldiers. When the first unfortunate enemy soldier died by your hand, his blood splashed onto your tunic. You quickly rushed to the next opponent with a deafening cry!;Vous avez combattu dans le feu de l'action aux c�t�s de vos soldats. Lorsque le premier malheureux soldat ennemi est mort par votre main, son sang a �clabouss� votre ...
+EVTDESC_WCCLS_105;[target_class.GetFirstName] fights in the heat of battle alongside [target_class.GetHerHis] soldiers. When the first unfortunate enemy soldier dies by [target_class.GetHerHis] hand, his blood splashes onto the tunic. [target_class.GetFirstName] quickly rushes to the next opponent with a deafening cry!;Vous avez combattu dans le feu de l'action aux c�t�s de vos soldats. Lorsque l ...
```
I don't get why you want to change that to speak of the character in 3rd person. It's an event about you yourself fighting, isn't it?

LordSilvermane replied:
I agree with Arithon, I like the first one more.

LordSilvermane replied:
I agree with Arithon, I like the first one more.

zumbak04 replied:
Changed in 194073d15e32b18f2c3edd36416b2e4a68072ae4.

arithon commented on localisation/00_zuwc_united_localisation.csv:
```diff
@@ -3058,23 +3058,23 @@ DYN12_DESC;With a score of $SCORE$ you$FAILED$ beat House Mechagon. In the dista
+soothing_mist;Wave Soothing Mist;;;;;;;;;;;;;x
+soothing_mist_desc;Wave soothing mist healing the stress and depression of �Y[Root.GetBestName]�!.;;;;;;;;;;;;;x
+text_effect_soothing_mist;Enables �GWave Soothing Mist�! on character portraits;;;;;;;;;;;;;x
+EVTDESC_WCCLS_1610;[target_class.GetFirstName] fights in the heat of battle alongside [target_class.GetHerHis] soldiers. After the first unfortunate enemy soldier dies by [target_class.GetHerHis] hand, [target_class.GetSheHe] calls three elemental spirits to aid in the battle. They immediately rush into combat, knocking the enemy back.;;;;;;;;;;;;;x
```
Is this how it worked in WoW? Because in WC3, he split himself into the 3 elementals...

LordSilvermane replied:
Arithon is right. 

> Split into 3 elemental spirits for 15 sec, each spirit dealing 45% of normal damage and healing. You directly control the Storm spirit, while Earth and Fire spirits mimic your attacks on nearby enemies.

zumbak04 replied:
The event together with localization is reworked in cc27490f6d5b72ba63e5736325b44cf3ae3edc73.

LordSilvermane commented on localisation/000_wc_traits.csv:
```diff
@@ -638,3 +638,25 @@ light_elune_branch;Follower of Mother Moon;;Anh
+class_monk_8;Master [GetMonkNameCap] VIII;;Meister-[GetMonkNameCap] VIII;;;;;;;;;;;x
+class_monk_9;Grand Master [GetMonkNameCap] IX;;Gro�meister-[GetMonkNameCap] IX;;;;;;;;;;;x
+class_monk_10;Legendary [GetMonkNameCap] X;;Legend�re[From.GetrMale] [GetMonkNameCap] X;;;;;;;;;;;x
+GetMonkDesc_desc;This character is a monk. Masters of bare-handed combat, monks never rely solely on the need to have a weapon in their hands to defend against their enemies. Monks bring a unique martial arts style to any fight and harness an exotic form of magical energy that�s unfamiliar to those, who practice other arcane arts.;;Dieser Charakter ist ein M�nch. Als Meister des waffenlosen Kampf ...
```
Removed unnecessary comma and apostrophe in last sentence. Added an `is` in place of the apostrophe.
```suggestion
GetMonkDesc_desc;This character is a monk. Masters of bare-handed combat, monks never rely solely on the need to have a weapon in their hands to defend against their enemies. Monks bring a unique martial arts style to any fight and harness an exotic form of magical energy that is unfamiliar to those who practice other arcane arts.;;Dieser Charakter ist ein M�nch. Als Meister des waffenlosen Kampfes verlassen sich M�nche niemals allein auf die Notwendigkeit, eine Waffe zur Verteidigung gegen ihre Feinde zur Hand zu haben. M�nche bringen einen einzigartigen Kampfkunststil in jeden Kampf ein und machen sich eine exotische Form der magischen Energie zunutze, die denjenigen, die andere arkane K�nste aus�ben, unbekannt ist.;;;;;;;;;;;x
```

LordSilvermane commented on localisation/00_zuwc_united_localisation.csv:
```diff
@@ -3058,23 +3058,23 @@ DYN12_DESC;With a score of $SCORE$ you$FAILED$ beat House Mechagon. In the dista
+A_MONK;A monk.;;Die eine[student.GetrMaleOpp][student.GetsMale] M�nch[student.GetsMale][studentGetinFemale].;;;;;;;;;;;x
+teach_yourself_monk;Train Monk Skills on your own;;�bt Euch als M�nch;;;;;;;;;;;x
+teach_yourself_monk_desc;Train monk skills on your own and become a monk.;;�bt Euch in den F�higkeiten eines M�nchs und werdet ein[From.GeteFemale] solche[From.GetrMale].;;;;;;;;;;;x
+EVTDESC_WCCLS_1600;You decide to start with meditation exercises. You have heard, that the most important thing is to control your breathing. You sit in the lotus position and begin to breathe in and out calmly. Your training will take some time indeed.;;Ihr beschlie�t, mit Meditations�bungen zu beginnen. Wie Ihr erfahren habt, ist das Wichtigste, die Atmung zu kontrollieren. Ihr nehmt die Lotuss ...
```
Removed an unnecessary comma.
```suggestion
EVTDESC_WCCLS_1600;You decide to start with meditation exercises. You have heard that the most important thing is to control your breathing. You sit in the lotus position and begin to breathe in and out calmly. Your training will take some time indeed.;;Ihr beschlie�t, mit Meditations�bungen zu beginnen. Wie Ihr erfahren habt, ist das Wichtigste, die Atmung zu kontrollieren. Ihr nehmt die Lotusstellung ein und beginnt, ruhig ein- und auszuatmen. Ihr Training wird in der Tat einige Zeit in Anspruch nehmen.;;;;;;;;;;;x
```

LordSilvermane commented on localisation/00_zuwc_united_localisation.csv:
```diff
@@ -3058,23 +3058,23 @@ DYN12_DESC;With a score of $SCORE$ you$FAILED$ beat House Mechagon. In the dista
+teach_yourself_monk;Train Monk Skills on your own;;�bt Euch als M�nch;;;;;;;;;;;x
+teach_yourself_monk_desc;Train monk skills on your own and become a monk.;;�bt Euch in den F�higkeiten eines M�nchs und werdet ein[From.GeteFemale] solche[From.GetrMale].;;;;;;;;;;;x
+EVTDESC_WCCLS_1600;You decide to start with meditation exercises. You have heard, that the most important thing is to control your breathing. You sit in the lotus position and begin to breathe in and out calmly. Your training will take some time indeed.;;Ihr beschlie�t, mit Meditations�bungen zu beginnen. Wie Ihr erfahren habt, ist das Wichtigste, die Atmung zu kontrollieren. Ihr nehmt die Lotuss ...
+EVTDESC_WCCLS_1700;You have finally mastered the very last thing: the martial arts of bare-handed combat. Your hands hurt, but you feel stronger. Of course, you understand your training isn't over yet. The path to excellence may take your entire life, but you are happy with today's result.;;Ihr habt endlich das Allerletzte gemeistert: die Kampfk�nste des waffenlosen Kampfs. Eure H�nde schmerzen,  ...
```
Changed the first sentence up a bit.
```suggestion
EVTDESC_WCCLS_1700;You have finally mastered the final skill of a monk: the martial arts of bare-handed combat. Your hands hurt, but you feel stronger. Of course, you understand your training isn't over yet. The path to excellence may take your entire life, but you are happy with today's result.;;Ihr habt endlich das Allerletzte gemeistert: die Kampfk�nste des waffenlosen Kampfs. Eure H�nde schmerzen, aber Ihr f�hlt Euch st�rker. Nat�rlich wisst Ihr, dass Euer Training noch nicht zu Ende ist. Der Weg zur Vortrefflichkeit mag Euer ganzes Leben dauern, aber mit dem heutigen Ergebnis seid Ihr zufrieden.;;;;;;;;;;;x
```

Nelpok commented:
Storm earth and fire event is really overpowered, killing 25% of the enemy army seems excesive, prob the thing that requires to be either reworked or changed for something else entirely (Because if you just leave the morale buff its would feel like a nerfed warrior)

arithon commented:
I just got one idea... Wouldn't using Storm, Earth and Fire help in escaping from prison?^^

zumbak04 commented:
> Storm earth and fire event is really overpowered, killing 25% of the enemy army seems excesive, prob the thing that requires to be either reworked or changed for something else entirely (Because if you just leave the morale buff its would feel like a nerfed warrior)
> Edit: Prob a different description of the event too? because it seems a shaman is fighting by "summoning 3 elemental spirits" instead of a monk

The ability is reworked in cc27490f6d5b72ba63e5736325b44cf3ae3edc73.

zumbak04 commented:
> * Healing infection using “Heal Wounds” ability requires the character to be either severely_injured or wounded.
> As Alvieck said in discord, there are events that can give the character an “infection” trait without severely_injured or wounded present. In that case it would be logical if infected is removed by “Heal Wounds” regardless of severely_injured or wounded present or not.

Changed in 82f87b083bceeea9e50138eb9d6d340948cdb02f.

> * “Wave Soothing Mist” ability doesn’t remove stressed or depressed if used on AI characters. Only 1 mana is deducted from the mana pool when the spell is cast on AI character;

Fixed in 24e4182dc429aa208305a977d03397df7d7ad194.

> Storm, Earth and Fire combat event fires two events, one event followed by another - event WCCLS.1610 and then WCCLS.1611. Is it how it is supposed to work?

Fixed in 7a058f2df55de1622fb4d900bcf0157ce63f40f2.
